### PR TITLE
Add a persistent class for PPS geometry as well as a database geometry builder

### PR DESCRIPTION
**Pull request by wpcarvalho**
#### PR description:

The proposed code will allow to save PPS geometry in the conditions database. It comprises:

- The persistent class PDetGeomDesc and related files
- The record VeryForwardIdealGeometryRecord
- The builder PPSGeometryBuilder and related files
- Configuration files geometryIdealPPSFromDD_201[6,7,8]_cfi.py with the xml description used by the builder to create the geometry conditions.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by civanch** [comment]
please test

File: CondTools/Geometry/plugins/PPSGeometryBuilder.cc [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    throw cms::Exception("PPSGeometryBuilder") << "PDetGeomDesc is empty, no geometry to save in the database.";
+  } else {
+    if (dbservice_.isAvailable()) {
+      dbservice_->writeOne(pdet, dbservice_->beginOfTime(), "VeryForwardIdealGeometryRecord");
+    } else {
+      throw cms::Exception("PoolDBService required.");
+    }
+  }
+
+  return;
```
Comment (cvuosalo) [inline CondTools/Geometry/plugins/PPSGeometryBuilder.cc:91]: This `return` statement is not needed.

**Comment by cvuosalo** [comment]
The file ` CondTools/Geometry/plugins/PPSGeometryBuilder.cc` will need to be migrated to DD4hep very soon. That can be done in a subsequent PR.

**Comment by cvuosalo** [comment]
+1

File: CondTools/Geometry/plugins/PPSGeometryBuilder.cc (hunk 1 above)
Comment (wpcarvalho) [reply CondTools/Geometry/plugins/PPSGeometryBuilder.cc:91]: Hi Carl, as it is geometry-approved now, should I remove the return statement and commit it again? Or leave as it is and correct when I migrate it do DD4hep?

Comment (cvuosalo) [reply CondTools/Geometry/plugins/PPSGeometryBuilder.cc:91]: If any of the other reviewers ask for changes, you can make this fix along with other changes. If nothing else changes in this PR, you can fix this little statement in a later PR.

**Comment by pohsun** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**